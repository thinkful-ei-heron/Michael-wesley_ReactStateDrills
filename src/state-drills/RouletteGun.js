import React, { Component } from 'react'

export class RouletteGun extends Component {
  static defaultProps = {
    bulletInChamber: 8
  };

  state = {
    chamber: null,
    spinningTheChamber: false
  }
  handleClick = arg => {
    this.setState({
      spinningTheChamber: true
  })
  
  }

componentDidMount(){
  let timeout = setTimeout(() => {
    let result = Math.floor(Math.random() * 8)+1;
    this.setState({
      chamber:result,
      spinningTheChamber: false
    })
  }, 2000 );

  this.setState({
    timeout
    })
}

  componentWillUnmount() {
    clearTimeout(this.state.timeout);
    this.setState({
      timeout: null
    })
  }

  render() {
    let status;
    if (this.state.spinningTheChamber){
      status='spinning the chamber and pulling the trigger! ...';
    } else (
      status = this.state.chamber === this.props.bulletInChamber ? 'BANG!!!' : "you're safe!!" 
    )

    return (
      <div>
        <p>{status}</p>
        <button type="button" onClick={this.handleClick}>Pull the trigger!</button>
      </div>
    )
}
}

export default RouletteGun
