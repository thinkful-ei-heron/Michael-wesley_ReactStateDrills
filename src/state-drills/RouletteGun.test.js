import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import RouletteGun from './RouletteGun';



describe('Roulette test', () => {
  it('renders without crashing', () => {
    const result = document.createElement('div');
    ReactDOM.render(<RouletteGun/>, result);
    ReactDOM.unmountComponentAtNode(result);
  })
  it('renders as expected', ()=>{
    expect(renderer.create(<RouletteGun />)
    .toJSON())
    .toMatchSnapshot()
  })

})
