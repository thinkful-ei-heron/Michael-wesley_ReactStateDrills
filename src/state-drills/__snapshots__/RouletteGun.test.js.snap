// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Roulette test renders as expected 1`] = `
<div>
  <p>
    you're safe!!
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    Pull the trigger!
  </button>
</div>
`;
